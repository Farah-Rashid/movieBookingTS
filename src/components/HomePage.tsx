import React from "react";
import axios from "axios";
import { useState, useEffect } from "react";
import MovieCard from "./MovieCard";
import { Container } from "../styles/Container.styled";
import { Cards } from "../styles/Cards.styled";
import { searchUrl, apiUrl } from '../GlobalConstants'
import PageNotFound from "./PageNotFound";

type Response = {
  results: MovieInfo[];
};

type MovieInfo = {
  poster_path: string;
  title: string;
  id: number;
};

const HomePage = () => {
  const [movies, setMovies] = useState<MovieInfo[]>([]);
  const [searchTerm, setSearchTerm] = useState<string>("");


  const getMovies = async (url: string) => {
    try {
      const response = await axios.get<Response>(url);
      const result = response.data.results;
      setMovies(result);
    } catch (e) {
      console.log("error", e)
    }

  }

  const submit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    if (searchTerm) {
      getMovies(`${searchUrl}&query=${searchTerm}`);
    } else {
      getMovies(apiUrl);
    }
  };
  useEffect(() => {
    getMovies(apiUrl);
  }, []);

  return (
    <>
      <Container>
        {" "}
        <h1>Book Tickets</h1>
        <form onSubmit={submit}>
          <input
            type="text"
            placeholder="search"
            value={searchTerm}
            onChange={(e) => setSearchTerm(e.target.value)}
          />
        </form>
      </Container>

      {movies.length > 0 ? (
        <Cards>
          <MovieCard movies={movies} />
        </Cards>
      ) : (
        <PageNotFound />
      )}
    </>
  );
}
export default HomePage;
