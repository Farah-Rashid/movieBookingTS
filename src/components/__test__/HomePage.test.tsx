import { render, screen, fireEvent } from '@testing-library/react';
import HomePage from '../HomePage';
const getMovieData = require('./__mock__/index');
const axios = require('axios');
jest.mock('axios');
const mockResponse = {
  data: {
    results: [
      {
        id: 507086,
        poster_path: "/kAVRgw7GgK1CfYEJq8ME6EvRIgU.jpg",
        title: "Jurassic World Dominion",
      },
      {
        id: 759175,
        poster_path: "/9pCoqX24a6rE981fY1O3PmhiwrB.jpg",
        title: "The Princess",
      },
      {
        id: 919355,
        poster_path: "/uSMJbYhaEpQtF9vkMhpgljc0CA4.jpg",
        title: "Dragon Knight",
      },
    ],
  }
}



describe("testing homepage", () => {
  test('returns path of the movie', async () => {
    jest.spyOn(axios, "get").mockResolvedValue({ data: mockResponse.data.results })
    const movieData = await getMovieData();
    expect(movieData[0].poster_path).toEqual("/kAVRgw7GgK1CfYEJq8ME6EvRIgU.jpg");
  });

  test('should take a snapshot', () => {
    const { container } = render(<HomePage />)
    expect(container).toMatchSnapshot();
  })

  test("testing input value", () => {
    render(<HomePage />)
    const inputValue = screen.getByPlaceholderText<HTMLInputElement>(/search/i)
    expect(inputValue).toBeInTheDocument()
    fireEvent.change(inputValue, { target: { value: "harry potter" } })
    expect(inputValue.value).toBe("harry potter");
  })
});