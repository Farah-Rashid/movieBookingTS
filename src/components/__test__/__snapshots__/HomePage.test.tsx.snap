// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing homepage should take a snapshot 1`] = `
<div>
  <header
    class="sc-hKMtZM cgCEkr"
  >
    <h1>
      Book Tickets
    </h1>
    <form>
      <input
        placeholder="search"
        type="text"
        value=""
      />
    </form>
  </header>
  <article
    class="sc-jSMfEi gciWvL"
  >
    <h2>
      Sorry, there is no result for keyword you searched.
    </h2>
    <img
      alt="no img found"
      src="undraw.png"
    />
  </article>
</div>
`;
